src/Resources/Products.php

```diff
@@ -54,4 +55,23 @@ public function get(string $productId): ProductResponse
 
         return new ProductResponse($response->json('data.product'));
     }
+
+    /**
+     * @throws FatalRequestException
+     * @throws RequestException
+     * @throws GraphQlException
+     * @throws JsonException
+     */
+    public function getByHandle(string $productHandle): ProductResponse
+    {
+        $response = $this->api->send(new GetProductByHandle($productHandle));
+
+        $this->validate($response);
+
+        if ($response->json('data.productByHandle') === null) {
+            throw new GraphQlException('Product not found');
+        }
+
+        return new ProductResponse($response->json('data.productByHandle'));
```

Comment:
Not sure the `GraphQlException` is the right exception. There's technically nothing wrong with the GraphQL query, it's just that no products where found.

What if it just returns a `ProductResponse` with no product? Not sure what the best approach to this is.

```suggestion
        return new ProductResponse($response->json('data.productByHandle') ?? []);
```